src-cljs/main.cljs

```diff
@@ -0,0 +1,32 @@
+(ns doctopus.main
+  (:require [goog.dom :as dom]
+            [doctopus.views.head-form :refer [head-form]]
+            [doctopus.views.tentacle-form :refer [tentacle-form]]
+            [doctopus.views.index :refer [main]]
+            [reagent.core :as reagent]))
+
+(enable-console-print!)
+
+(def pages {:add-head head-form
+            :add-tentacle tentacle-form
+            :edit-tentacle tentacle-form
+            :index main})
+
+(defn- get-app-state
+  []
+  (-> (dom/getElement "app-state")
+      (.-textContent)
+      (js/JSON.parse)
+      (js->clj :keywordize-keys true)))
+
+(defn- get-page-component
+  [app-state]
+  [((keyword (:page app-state)) pages) app-state])
+
+(defn init
+  [app-state]
+    (reagent/render-component
+     (get-page-component app-state)
+     (dom/getElement "app-content")))
```

Comment:
This is basically the equivalent of the `load-bundle` script from airship-ui. It's maybe the best option short of re-architecting this to be a single-page app.